buildtools/maven/maven.go

```diff
@@ -59,53 +72,141 @@ func (m *Maven) Compile(dir string) error {
 	return err
 }
 
-func Modules(dir string) ([]string, error) {
+// ResolveManifestFromBuildTarget tries to determine what buildTarget is supposed to be and then reads the POM
+// manifest file pointed to by buildTarget if it is a path to such a file or module.
+func (m *Maven) ResolveManifestFromBuildTarget(buildTarget string) (*Manifest, error) {
+	var pomFile string
+	if !strings.HasSuffix(buildTarget, ".xml") {
+		pomFile = filepath.Join(buildTarget, "pom.xml")
+	} else {
+		pomFile = buildTarget
+	}
+	if ok, _ := files.Exists(pomFile); !ok {
+		if strings.Count(buildTarget, ":") == 1 {
+			// This is likely a module ID.
+			return nil, fmt.Errorf("build target %q appears to be a module identifier", buildTarget)
```

Comment:
To a user, what does this error exactly mean? I’m not sure I would know what course of action to take after receiving an error like this

Reply:
fixed